state/state.go

```diff
@@ -38,9 +39,11 @@ type State struct {
 	mtx sync.Mutex
 	db  dbm.DB
 
-	// should not change
-	GenesisDoc *types.GenesisDoc
-	ChainID    string
+	// cachedGenesisDoc is the memoized genesisDoc to cut down
```

Comment:
I think this comment should go into git commit msg

Reply:
I'd rather keep it though as a mention for the future and purpose. Without it I feel like we'll be asking ourselves why we didn't just expose it as `GenesisDoc`, and also it is already mentioned in the git commit message :)